runtime/near-vm-logic/src/logic.rs

```diff
@@ -554,8 +492,15 @@ impl<'a> VMLogic<'a> {
     ///
     /// If the size of the registers exceed the set limit `MemoryAccessViolation`.
     pub fn random_seed(&mut self, register_id: u64) -> Result<()> {
-        let Self { context, registers, config, .. } = self;
-        Self::internal_write_register(registers, config, register_id, &context.random_seed)
+        let Self { context, registers, gas_counter, config, .. } = self;
+        gas_counter.pay_base(config.runtime_fees.ext_costs.random_seed)?;
+        Self::internal_write_register(
```

Comment:
For functions, like `random_seed`, `predecessor_account_id`, etc that take a `register_id` and write some blob of data into it we need to be consistent, and do either of the following three for all of them:
1. Do not charge different function-specific base fee, and do not charge different function-specific fee per byte, since we already charge base fee and fee per byte in `internal_write_register` these functions have little additional overhead over `internal_write_register` since they just read the data from `context`;
2. Charge function-specific base fee, but also keep charging inside `internal_write_register`. Even though these functions currently just read from `context` and pass it to `internal_write_register` they might be retrieving information differently in the future (e.g. computing it on the fly), so we can make it future proof by introducing function-specific base fee for now.
3. Charge function-specific base fee and function-specific fee per byte, in addition to charging inside `internal_write_register`. The same argument about being future-proof as in (2).

I suggest we go with (3)

Reply:
Yes, I'm following the 3rd one.